src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -713,18 +713,37 @@ public static unsafe double Round(double value, int digits, MidpointRounding mod
 
                 value *= power10;
 
-                if (mode == MidpointRounding.AwayFromZero)
+                if (mode is MidpointRounding.ToEven)
                 {
-                    var fraction = ModF(value, &value);
+                    // Rounds to the nearest value; if the number falls midway,
+                    // it is rounded to the nearest value with an even least significant digit
+                    value = Round(value);
+                }
+                else if (mode is MidpointRounding.AwayFromZero)
+                {
+                    // Rounds to the nearest value; if the number falls midway,
+                    // it is rounded to the nearest value above (for positive numbers) or below (for negative numbers)
+                    double fraction = ModF(value, &value);
 
                     if (Abs(fraction) >= 0.5)
                     {
                         value += Sign(fraction);
                     }
                 }
-                else
+                else if (mode is MidpointRounding.ToZero)
                 {
-                    value = Round(value);
+                    // Directed rounding: Round toward zero, to nearest value above (positive numbers) or below (negative numbers)
+                    value = Truncate(value);
+                }
+                else if (mode is MidpointRounding.ToNegativeInfinity)
+                {
+                    // Directed Rounding: Round down to the next value, toward −∞
```

Comment:
I don't think we typically put non-standard symbols in source code, as some editors/viewers don't show these correctly.  Can we just put `negative infinity` instead?